### PR TITLE
add card

### PR DESCRIPTION
**Pull request by dimarogkov**
https://dimarogkov.github.io/layout_product-cards
https://dimarogkov.github.io/layout_product-cards/report/html_report

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -1,13 +1,60 @@
...
+  </head>
+  <body>
+    <div class="card" data-qa="card">
+      <!-- img -->
+      <div class="card__img">
+        <picture>
+          <img
+            src="images/imac.jpeg"
+            alt="card-img"
+            loading="lazy"
```
Comment (Viktor-Kost) [inline src/index.html:19]: Use a more descriptive 'alt' attribute value for the image, such as 'iMac 27" Retina 5K'.

File: src/style.css [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  position: relative;
+  width: 160px;
+  height: 0;
+  padding-bottom: 134px;
+  background-color: var(--color-grey);
+  margin: 0 auto 40px;
+}
+
+.card__img img {
+  position: absolute;
```
Comment (Viktor-Kost) [inline src/style.css:54]: There is no need to use position: absolute for the img element. Remove position, top, and left properties and adjust other properties if needed.

File: src/style.css [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+.card__btn:active {
+  transform: translateY(5px);
+}
+
+.card__img:last-child,
+.card__head:last-child,
+.card__title:last-child,
+.card__stars:last-child,
+.card__price:last-child {
+  margin-bottom: 0;
```
Comment (Viktor-Kost) [inline src/style.css:160]: These last-child styles might not be necessary. Check if your layout still works without them and remove them if they are not needed.

File: src/style.css [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+  padding: 32px 16px 16px;
+  border: 1px solid #f3f3f3;
+  border-radius: 5px;
+  background-color: var(--color-white);
+}
+
+.card__img {
+  position: relative;
+  width: 160px;
+  height: 0;
```
Comment (Viktor-Kost) [inline src/style.css:47]: The height of .card__img should not be 0. Set the height to the desired size and remove padding-bottom.

File: src/index.html [hunk 5]
```diff
@@ -1,13 +1,60 @@
...
+    <meta charset="UTF-8">
+    <meta name="viewport"
+          content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
+    <title>Product cards</title>
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@100;300;400;500;700;900&display=swap" rel="stylesheet">
+    <link rel="stylesheet" href="./style.css">
+  </head>
+  <body>
+    <div class="card" data-qa="card">
+      <!-- img -->
```
Comment (Viktor-Kost) [inline src/index.html:13]: remove comments

File: src/index.html [hunk 6]
```diff
@@ -1,13 +1,60 @@
...
+      <div class="card__img">
+        <picture>
+          <img
```
Comment (Viktor-Kost) [inline src/index.html:16]: 1) no needs to create img wrapper such as div
2) it's not a right case to use picture tag
3) just use tag img with a class `card__img`

File: src/index.html [hunk 7]
```diff
@@ -1,13 +1,60 @@
...
+            src="images/imac.jpeg"
+            alt="card-img"
+            loading="lazy"
+          >
+        </picture>
+      </div>
+
+      <!-- cont -->
+      <div class="card__cont">
```
Comment (Viktor-Kost) [inline src/index.html:25]: redundant wrapper
```suggestion
```

File: src/index.html [hunk 8]
```diff
@@ -1,13 +1,60 @@
...
+        <!-- head -->
+        <div class="card__head">
```
Comment (Viktor-Kost) [inline src/index.html:27]: redundant wrapper
```suggestion
```

File: src/index.html [hunk 9]
```diff
@@ -1,13 +1,60 @@
...
+          <h3 class="card__title">APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)</h3>
+          <div class="card__code">Product code: 195434</div>
```
Comment (Viktor-Kost) [inline src/index.html:29]: use semantic tags
```suggestion
          <p class="card__code">Product code: 195434</p>
```

File: src/index.html [hunk 10]
```diff
@@ -1,13 +1,60 @@
...
+            <span class="stars__star"></span>
+          </div>
+
+          <p>Reviews: 5</p>
+        </div>
+
+        <!-- price -->
+        <div class="card__price">
+          <p>Price:</p>
+          <strong>$2,199</strong>
```
Comment (Viktor-Kost) [inline src/index.html:48]: just `<p>`
```suggestion
          <p>$2,199</p>
```

File: src/index.html [hunk 11]
```diff
@@ -1,13 +1,60 @@
...
+        </div>
+
+        <!-- btn -->
+        <a
+          href="#"
+          class="card__btn"
+          data-qa="hover"
+        >Buy</a>
```
Comment (Viktor-Kost) [inline src/index.html:56]: follow code styling recommendations
```suggestion
        <a
          href="#"
          class="card__btn"
          data-qa="hover"
        >
          Buy
        </a>
```

File: src/style.css [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+  justify-content: space-between;
+  width: 100%;
+  margin-bottom: 24px;
+}
+
+.card__price {
+  margin-bottom: 16px;
+}
+
+.card__stars p {
```
Comment (Viktor-Kost) [inline src/style.css:101]: is't better not select by tag. Use classes

File: src/style.css [hunk 13]
```diff
@@ -0,0 +1,191 @@
...
+  position: relative;
+  top: 1px;
```
Comment (Viktor-Kost) [inline src/style.css:103]: 1) top: 1px; is redundant
2) use styles fro stars from a previous task
```suggestion
```

File: src/style.css [hunk 14]
```diff
@@ -0,0 +1,191 @@
...
+  text-transform: uppercase;
+}
+
+.card__btn {
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: 40px;
```
Comment (Viktor-Kost) [inline src/style.css:132]: it's better not hardcode heigh for elements with content inside. Use padding

File: src/style.css [hunk 15]
```diff
@@ -0,0 +1,191 @@
...
+
+.stars__star {
+  position: relative;
+  display: block;
+  width: 16px;
+  height: 16px;
+  background-image: url("images/star.svg");
+  background-repeat: no-repeat;
+  background-size: auto;
+  background-position: center;
```
Comment (Viktor-Kost) [inline src/style.css:178]: apply background `shorthand`

File: src/style.css [hunk 16]
```diff
@@ -0,0 +1,191 @@
...
+.card__img:last-child,
+.card__head:last-child,
+.card__title:last-child,
+.card__stars:last-child,
+.card__price:last-child {
+  margin-bottom: 0;
+}
+
+/* stars */
+.stars {
```
Comment (Viktor-Kost) [inline src/style.css:164]: it'd be nice if you separate blocks such as card and stars to their own scss files

File: src/style.css [hunk 17]
```diff
@@ -0,0 +1,191 @@
...
+  font-weight: 700;
+  line-height: 16px;
+  color: var(--color-white);
+  text-transform: uppercase;
+  padding: 0 20px;
+  border-radius: 5px;
+  border: 1px solid var(--color-blue);
+  background-color: var(--color-blue);
+  will-change: transform;
+  transition: all 0.4s;
```
Comment (Viktor-Kost) [inline src/style.css:143]: specify what exactly you'd like to change by adding transition-property

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

File: src/style.css [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+.card__reviews {
+  position: relative;
+  top: 1px;
+  font-size: 10px;
+  line-height: 14px;
+  color: var(--color-black);
+  white-space: nowrap;
+}
+
+.card__price p {
```
Comment (YegorKorenkov) [inline src/style.css:95]: Better to provide some class to element, same for .card__price p:last-child

File: src/style.css [hunk 19]
```diff
@@ -0,0 +1,162 @@
...
+  display: block;
+  width: 160px;
+  height: 134px;
+  background-color: var(--color-grey);
+  margin: 0 auto 40px;
+}
+
+.card__title {
+  position: relative;
+  width: 100%;
```
Comment (YegorKorenkov) [inline src/style.css:55]: You don't need to set width: 100%, please remove and check for all cases

File: src/style.css (hunk 13 above)
Comment (YegorKorenkov) [reply src/style.css:88]: Not fixed

File: src/style.css (hunk 17 above)
Comment (YegorKorenkov) [reply src/style.css:125]: Not fixed, please add smth like this:
transition: background-color 0.4s;
read more: https://developer.mozilla.org/en-US/docs/Web/CSS/transition

File: src/style.css [hunk 20]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+.card__btn {
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: 40px;
+  font-size: 14px;
```
[the remark this answers is not in the document]
Comment (YegorKorenkov) [reply src/style.css:115]: font-size 14px - default browser rule, please remove

**Review by YegorKorenkov — CHANGES_REQUESTED** [review]

File: src/style.css [hunk 21]
```diff
@@ -0,0 +1,157 @@
...
+p {
+  margin: 0;
+}
+
+a {
+  text-decoration: none;
+  color: var(--color-black);
+}
+
+/* card */
```
Comment (OleziO) [inline src/style.css:33]: Remo all comments,

**Review by OleziO — CHANGES_REQUESTED** [review]

**Review by etojeDenys — APPROVED** [review]
lgtm

**Pull request closed without merging**